lz4: Update to 1.9.4

chausner opened the pull request:
Specify library name and version:  **lz4/1.9.4**

This adds the latest patch release of lz4.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

chausner commented:
With version 1.9.4, lz4 also installs a CMake package: https://github.com/lz4/lz4/pull/1030/files. They use the target namespace [LZ4::](https://github.com/lz4/lz4/pull/1030/files#diff-5bae9682a66f5788780b37b57c2a4fe68ce47a1907dd20eb84e12a745e36d9b0R229) while Conan has been using lz4::. Is this a problem? Do we need to adapt anything in the recipe?

chausner commented:
Not sure why the test package fails in CI, it works fine when I test it locally on Linux and gcc. Does anybody spot the problem?

uilianries commented:
Just restarted the job now. I can't reproduce locally using same docker image.

uilianries commented:
@chausner I'm not sure about that error, but I updated some parts of the test package folders. Please, consider https://github.com/chausner/conan-center-index/pull/1.

chausner commented:
> @chausner I'm not sure about that error, but I updated some parts of the test package folders. Please, consider [chausner#1](https://github.com/chausner/conan-center-index/pull/1).

Thank you! I merged it.

chausner commented:
> @chausner I'm not sure about that error, but I updated some parts of the test package folders. Please, consider [chausner#1](https://github.com/chausner/conan-center-index/pull/1).

I think I found the problem and fixed it in https://github.com/conan-io/conan-center-index/pull/12895/commits/799d8f3feacdfc460b6757ea1b42bc7d47938e6c.

uilianries reviewed (approved):
what mess those cmake targets! Thank you for managing it!

SpaceIm commented on recipes/lz4/all/conanfile.py:
```diff
@@ -79,17 +80,48 @@ def build(self):
+            # TODO: to remove in conan v2 once legacy generators removed
+            self._create_cmake_module_alias_targets(
+                os.path.join(self.package_folder, self._module_file_rel_path),
+                {"lz4": "LZ4::lz4" if self.options.shared else "LZ4::lz4_static"}
```
??

chausner replied:
Yes, I copied it from other recipes but have yet to learn what the purpose is and when to use this. Is there documentation on functions like these somewhere?

What would be the proper fix here? Completely removing the function call because it is not needed?

SpaceIm replied:
It's needed, I've edited my review with a code suggestion.

SpaceIm replied:
well too late, it has been merged, I'll open a PR

SpaceIm replied:
https://github.com/conan-io/conan-center-index/pull/12953

SpaceIm commented on recipes/lz4/all/conanfile.py:
```diff
@@ -79,17 +80,48 @@ def build(self):
+
     def package_info(self):
         self.cpp_info.set_property("pkg_config_name", "liblz4")
+        if Version(self.version) >= "1.9.4":
+            self.cpp_info.set_property("cmake_file_name", "lz4")
+            self.cpp_info.set_property("cmake_target_name", "LZ4::lz4" if self.options.shared else "LZ4::lz4_static")
```
not sure it's a good idea to have this condition.

chausner replied:
Also here I am lacking the necessary insight. Let me know if I should remove the condition or change it in any way.

SpaceIm commented on recipes/lz4/all/conanfile.py:
```diff
@@ -79,17 +80,48 @@ def build(self):
+
     def package_info(self):
         self.cpp_info.set_property("pkg_config_name", "liblz4")
+        if Version(self.version) >= "1.9.4":
+            self.cpp_info.set_property("cmake_file_name", "lz4")
+            self.cpp_info.set_property("cmake_target_name", "LZ4::lz4" if self.options.shared else "LZ4::lz4_static")
```
```suggestion
        if Version(self.version) >= "1.9.4":
            self.cpp_info.set_property("cmake_file_name", "lz4")
            self.cpp_info.set_property("cmake_target_name", "LZ4::lz4" if self.options.shared else "LZ4::lz4_static")
```

`LZ4::lz4` if shared? Are you sure? Looking at source code it's `LZ4::lz4_shared`: https://github.com/lz4/lz4/blob/v1.9.4/build/cmake/CMakeLists.txt#L105

chausner replied:
Good catch, you're right.
